spec/models/doorkeeper/access_token_spec.rb

```diff
@@ -143,6 +143,12 @@ module NoGenerate
         subject.resource_owner_id = nil
         expect(subject).to be_valid
       end
+
+      it 'is valid without application_id' do
+        # For resource owner credentials flow
+        subject.application_id = nil
+        expect(subject).to be_valid
+      end
```

Comment:
I see. There is actually no conditional presence validation for this, so it doesn't make sense in it's current form anyway.